examples/spans.rs

```diff
@@ -0,0 +1,29 @@
+use fuel_telemetry::prelude::*;
+
+fn main() {
```

Comment:
i don't really understand the point of this example....

how did you envision span's being used in practice?

Reply:
Examples have been updated, which now show practical telemetry being recorded, and how the functions differ between the different files.